src/core/public/application/types.ts

```diff
@@ -542,6 +542,8 @@ export interface ApplicationSetup {
     contextName: T,
     provider: IContextProvider<AppMountDeprecated, T>
   ): void;
+
+  currentAppId$: Observable;
```

Comment:
This should have a generic type. Also should this be exposed during setup? No applications are mounted during setup, so I think we can remove this.